### ISSUE TITLE
Add Gutenberg extension -specific filter for availability

### ISSUE THREAD

**Issue by simison** [Automattic/jetpack#12028]
Add a filter to easily disable availability for specific blocks.

Current filter that allows modifying the whole availability array seems little fragile and can break if we change something at our end:

```php
add_filter( 'jetpack_set_available_blocks', 'jetpack_unregister_mailchimp_block' );
function jetpack_unregister_mailchimp_block( $blocks ) {
    $found = array_search( 'mailchimp', $blocks );
    if ( false !== $found ) {
        unset( $blocks[ $found ] );
    }
    return $blocks;
}
```

Having a simple filter for each block `jetpack_extension_{extension-slug}_availability` that's just a boolean would be straightforward for partners to use to disable specific block-editor integrations:

```php
add_filter( 'jetpack_extension_mailchimp_availability', '__return_false' );
```

convo p1555011362177100-slack-jetpack & p7fD6U-2q5-p2

**Comment by ebinnion** [comment]
> Current filter that allows modifying the whole availability array seems little fragile and can break if we change something at our end

To clarify, what do you mean by break?

As long as we're passing an array to the filter, I wouldn't anticipate breakage in the sense of a white screen. But, if we changed how we use that filter, then I would expect breakage in the sense that the filter no longer works as expected.

The same could be said if we got specific with the filter, such as `jetpack_extension_{extension-slug}_availability`. If we change how we use that filter, then we're still prone to breakage in the sense that it may not work the same way.

**Comment by simison** [comment]
Good question!

I had two aspects in mind specifically:

- We might change code so that it expects specific block handles to be available in availability array. (prolly won't happen)
- Users of this filter might not use clean `unset()` to modify the array, but adjust settings in it directly. We might change the footprint of the content later on in which case this would break. (I'm more worried about this)

I felt it would be safer not to pass the whole footprint of the availability data to filter and just target what consumers really need (an "available" boolean). Something like that can't really be used in a way that it would break anything even if we moved code around at our end.

I don't think we necessarily need to add this, especially after @lezama testing that `unset()` doesn't currently have negative consequences. I mainly posted it here to record an idea that came up during the convo.

I labelled it as "question" to emphasize that better. :-)
